File: web/lib/arena/domains.ts
// Values must match ArenaDomain in runner/src/coval_bench/api/schemas.py — they are
// stored on battles and used as leaderboard keys ("all" is reserved for the aggregate board).
export const ARENA_DOMAINS = [
  { value: "customer-service", label: "Customer Service" },
  { value: "healthcare", label: "Healthcare" },
  { value: "sales", label: "Sales" },
  { value: "receptionist-booking", label: "Receptionist / Booking" },
  { value: "other", label: "Other" },
] as const;

export type ArenaDomain = (typeof ARENA_DOMAINS)[number]["value"];

export function isArenaDomain(value: unknown): value is ArenaDomain {
  return ARENA_DOMAINS.some((d) => d.value === value);
}

File: runner/src/coval_bench/arena/domains.py
# Copyright 2026 The Coval Benchmarks Authors
# SPDX-License-Identifier: Apache-2.0

"""The closed set of domains a battle can be tagged with.

Lives here rather than in ``api.schemas`` because it is arena vocabulary, not an API
shape: ``arena.prompts`` needs it, and importing it from ``api`` pulled in the whole
FastAPI app, which cycles back through ``api.routers.arena``.
"""

from __future__ import annotations

from typing import Literal

ArenaDomain = Literal["customer-service", "healthcare", "sales", "receptionist-booking", "other"]
"""Each value doubles as a leaderboard key, so the set is closed and excludes ``all`` —
that key is reserved for the aggregate board."""
